Sources/eppo/RuleEvaluator.swift

```diff
@@ -3,131 +3,259 @@ import Semver
+                    let conditionValueStr = try? condition.value.getStringValue()
+                    if let valueVersion = valueStr.flatMap(Semver.init), let conditionVersion = conditionValueStr.flatMap(Semver.init) {
                         // If both strings are valid Semver strings, perform a Semver comparison
-                        switch condition.targetingOperator {
-                        case .GreaterThanEqualTo:
+                        switch condition.operator {
+                        case .greaterThanEqual:
                             return valueVersion >= conditionVersion
-                        case .GreaterThan:
+                        case .greaterThan:
                             return valueVersion > conditionVersion
-                        case .LessThanEqualTo:
+                        case .lessThanEqual:
                             return valueVersion <= conditionVersion
-                        case .LessThan:
+                        case .lessThan:
                             return valueVersion < conditionVersion
                         default:
                             throw Errors.UnexpectedValue
                         }
                     } else {
                         // If either string is not a valid Semver, fall back to double comparison
-                        let valueDouble = try value.doubleValue()
-                        let conditionDouble = try condition.value.doubleValue()
-                        switch condition.targetingOperator {
-                        case .GreaterThanEqualTo:
+                        let valueDouble = try value.getDoubleValue()
+                        let conditionDouble = try condition.value.getDoubleValue()
+                        switch condition.operator {
+                        case .greaterThanEqual:
                             return valueDouble >= conditionDouble
-                        case .GreaterThan:
+                        case .greaterThan:
                             return valueDouble > conditionDouble
-                        case .LessThanEqualTo:
+                        case .lessThanEqual:
                             return valueDouble <= conditionDouble
-                        case .LessThan:
+                        case .lessThan:
                             return valueDouble < conditionDouble
                         default:
                             throw Errors.UnexpectedValue
                         }
                     }
-                } catch {
+                } catch let e {
                     // If stringValue() or doubleValue() throws, or Semver creation fails
                     return false
                 }
-            case .Matches:
+            case .matches:
                 return try Compare.compareRegex(
-                    value.stringValue(),
-                    condition.value.stringValue()
+                    value.getStringValue(),
```

Comment:
I think you need to change the applied function to `matchesRegex()` as we landed on partial matches by default with regular expressions.

Reply:
@schmit would modifying the test files be possible here as well?

Reply:
the current implementation:

```
return a.range(of: pattern, options:.regularExpression) != nil;
```

already supports partial matching. I will rename the wrapper method for clarity.